### PR TITLE
Add script to copy distributed commits to stable branch

### PR DESCRIPTION
**Pull request by bigfootjon**
This will be used as part of a prototype of a stable pytorch with a fast-moving distributed folder

Tasks: T189915739

Test plan:

I ran the script in a few configurations on my local machine. It worked as expected

### CODE REVIEW

File: .github/scripts/sync_distributed_folder_prototype.sh [hunk 1]
```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
+
+ORIG_BRANCH=$(git rev-parse --abbrev-ref HEAD)
+git checkout fbcode/pytorch-stable-prototype
+for SHA in $(git log 4333e122d4b74cdf84351ed2907045c6a767b4cd..main --pretty="%h" --reverse -- torch/distributed torch/csrc/distributed test/distributed test/cpp/c10d benchmarks/distributed)
+do
+    if git merge-base --is-ancestor $SHA HEAD || [[ $(git log --grep="(cherry picked from commit $SHA") ]]
```
Comment (wconstab) [inline .github/scripts/sync_distributed_folder_prototype.sh:7]: im not familiar enough with git merge-base, what is the logic here?

Comment (wconstab) [reply .github/scripts/sync_distributed_folder_prototype.sh:7]: i think its saying if `SHA` is already present on the stable branch, don't cherry pick it?

Comment (bigfootjon) [reply .github/scripts/sync_distributed_folder_prototype.sh:7]: Yeah that's basically it, I'll add a comment

**Review by seemethere — APPROVED** [review]

**Review by malfet — APPROVED** [review]
LGTM, but please consider replacing shell script with a Python one

**Comment by bigfootjon** [comment]
@pytorch merge -f "CI failures are unrelated"

**Pull request closed without merging**